src/ring_theory/finiteness.lean

```diff
@@ -182,8 +188,64 @@ begin
     exact finite_type.of_surjective (finite_type.mv_polynomial R (fin n)) f hsur }
 end
 
+/-- A finitely presented algebra is of finite type. -/
+lemma of_finitely_presented : finitely_presented R A → finite_type R A :=
+begin
+  rintro ⟨n, ⟨f, hf⟩⟩,
+  apply (finite_type.iff_quotient_mv_polynomial'').2,
+  use [n, f, hf.1]
+end
+
 end finite_type
 
+namespace finitely_presented
+
+theorem spec (h : finitely_presented R A) :
+  ∃ (n : ℕ) (f : mv_polynomial (fin n) R →ₐ[R] A),
+  surjective f ∧ (ring_hom.ker (alg_hom.to_ring_hom f)).fg :=
+h
+
+/-- If `e : A ≃ₐ[R] B` and `A` is finitely presented, then so is `B`. -/
+lemma equiv (hfp : finitely_presented R A) (e : A ≃ₐ[R] B) : finitely_presented R B :=
+begin
+  obtain ⟨n, f, hf⟩ := spec R A hfp,
```

Comment:
What's the purpose of `spec` here? This line would work as follows I think?
```suggestion
  obtain ⟨n, f, hf⟩ := hfp,
```

If you want to keep `spec`, I'd be inclined to make the `R` and `A` args implicit so that you can just use `h.spec`.

Reply:
Ah yes, this is because `finitely_presented` was a class. I decided to change it and this is now useless. (I can make it a structure if you think it is better).